### PR TITLE
add redis cache reboot/regenerate key

### PR DESCRIPTION
**Pull request by yungezz**

### CODE REVIEW

File: library/azure_rm_rediscache.py [hunk 1]
```diff
@@ -109,6 +109,28 @@
...
+                default: all
+    regenerate_key:
+        description:
+            - Regenerate Redis cache's access keys.
+        suboptions:
+            key_type:
+                description: The Redis key to regenerate.
+                choices:
+                    - primary
+                    - Secondary
```
Comment (zikalino) [inline library/azure_rm_rediscache.py]: should be lowercase?

File: library/azure_rm_rediscache.py [hunk 2]
```diff
@@ -570,6 +657,41 @@ def get_rediscache(self):
...
+        :return: True
+        '''
+        self.log("Force reboot the redis cache instance {0}".format(self.name))
+        try:
+            response = self._client.redis.force_reboot(resource_group_name=self.resource_group,
+                                                       name=self.name,
+                                                       reboot_type=self.reboot['reboot_type'],
+                                                       shard_id=self.reboot.get('shard_id'))
+        except CloudError as e:
+            self.log('Error attempting to force reboot the redis cache instance.')
```
Comment (zikalino) [inline library/azure_rm_rediscache.py]: i wonder if we really need log and error

Comment (yungezz) [reply library/azure_rm_rediscache.py]: why not? if user specify reboot, they should be able to know what happened if reboot failed.

Comment (zikalino) [reply library/azure_rm_rediscache.py]: so error will do that :-)
well, not important anyway....

**Pull request merged**